primefaces/src/main/java/org/primefaces/util/FileUploadUtils.java

```diff
@@ -242,8 +242,8 @@ private static boolean isValidFileContent(PrimeApplicationContext context, FileU
             return true;
         }
 
-        String tempFilePrefix = UUID.randomUUID().toString();
-        Path tempFile = Files.createTempFile(tempFilePrefix, null);
+        Path tempDirectory = Files.createTempDirectory("pf-fileupload");
+        Path tempFile = Files.createTempFile(tempDirectory, UUID.randomUUID().toString(), null);
```

Comment:
is this tmp directory really necessary?

Reply:
I kept it because when the second pass is run I completely keep the original filename received from the upload and wanted to avoid "collisions" when two files with the same name are uploaded at exactly the same time, creating/overriding something like `/tmp/file.jpg` twice

Reply:
Since your tempfile name will be "randomly" generated, it is very unlikely to have same name

Reply:
Sounds good, PR is updated